dev/ScrollViewer/ScrollViewer.cpp

```diff
@@ -483,11 +505,127 @@ void ScrollViewer::OnGotFocus(winrt::RoutedEventArgs const& args)
         m_focusInputDeviceKind == winrt::FocusInputDeviceKind::Mouse ||
         m_focusInputDeviceKind == winrt::FocusInputDeviceKind::Pen;
 
-    ShowIndicators();
+    UpdateScrollControllersAutoHiding();
+    if (AreScrollControllersAutoHiding())
+    {
+        UpdateVisualStates(true /*useTransitions*/, true /*showIndicators*/);
+    }
 }
 
 #pragma endregion
 
+bool ScrollViewer::AreScrollControllersAutoHiding()
+{
+    // Use the cached value unless it was invalidated.
+    if (s_autoHideScrollControllersValid)
+    {
+        return s_autoHideScrollControllers;
+    }
+
+    s_autoHideScrollControllersValid = true;
+
+    if (SharedHelpers::IsRS4OrHigher())
+    {
+        com_ptr<ScrollViewerTestHooks> globalTestHooks = ScrollViewerTestHooks::GetGlobalTestHooks();
+
+        if (globalTestHooks)
+        {
+            winrt::IReference<bool> autoHideScrollControllers = globalTestHooks->AutoHideScrollControllers();
+
+            if (autoHideScrollControllers)
+            {
+                // Test hook takes precedence over UISettings and registry key settings.
+                s_autoHideScrollControllers = autoHideScrollControllers.Value();
+                return s_autoHideScrollControllers;
+            }
+        }
+    }
+
+    if (s_uiSettings5)
+    {
+        // Use the 19H1+ UISettings property.
+        s_autoHideScrollControllers = s_uiSettings5.AutoHideScrollBars();
+    }
+    else if (SharedHelpers::IsRS4OrHigher())
+    {
+        // Use the RS4+ registry key HKEY_CURRENT_USER\Control Panel\Accessibility\DynamicScrollbars
+        s_autoHideScrollControllers = RegUtil::UseDynamicScrollbars();
+    }
+    else
+    {
+        // ScrollBars are auto-hiding prior to RS4.
+        s_autoHideScrollControllers = true;
+    }
+
+    return s_autoHideScrollControllers;
+}
+
+void ScrollViewer::OnAutoHideScrollBarsChanged(
+    winrt::UISettings const& uiSettings,
+    winrt::UISettingsAutoHideScrollBarsChangedEventArgs const& args)
+{
+    MUX_ASSERT(SharedHelpers::IsRS4OrHigher());
```

Comment:
>IsRS4OrHigher [](start = 30, length = 13)

I assume this should be 19H1+ since this event would only be raised for 19H1+, right?

Reply:
Yes indeed. Changed it to 19H1.